### PR TITLE
Fixing #1231. Adding order and status to trace query.

### PR DESCRIPTION
**Pull request by ajanthan**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [ *] New feature provided
- [ ] Improve performance

- Related issues
#1231 
___
### Bug fix
- Bug description.

- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.
Added status and order to trace query API and updated [UI](https://github.com/apache/incubator-skywalking-ui/pull/172)

### CODE REVIEW

**Comment by ajanthan** [comment]
OK. Let me change Ints to enum and resend the pull request.

**Comment by wu-sheng** [comment]
Thanks and look forward to have this soon.

**Comment by ajanthan** [comment]
Updated the pull request as requested.

**Comment by wu-sheng** [comment]
@ajanthan Your CI  fails.

File: apm-collector/apm-collector-storage/collector-storage-es-provider/src/main/java/org/apache/skywalking/apm/collector/storage/es/dao/ui/SegmentDurationEsUIDAO.java [hunk 1]
```diff
@@ -73,16 +75,30 @@ public TraceBrief loadTop(long startSecondTimeBucket, long endSecondTimeBucket,
             boolQueryBuilder.must().add(rangeQueryBuilder);
         }
         if (StringUtils.isNotEmpty(operationName)) {
-            mustQueryList.add(QueryBuilders.matchPhraseQuery(SegmentDurationTable.SERVICE_NAME.getName(), operationName));
+            mustQueryList.add(QueryBuilders.matchQuery(SegmentDurationTable.SERVICE_NAME.getName(), operationName));
```
[the remark this answers is not in the document]
Comment (wu-sheng) [reply apm-collector/apm-collector-storage/collector-storage-es-provider/src/main/java/org/apache/skywalking/apm/collector/storage/es/dao/ui/SegmentDurationEsUIDAO.java]: `matchPhraseQuery` is from #1194 . @ajanthan Why do you use `matchQuery`?

[the remark this answers is not in the document]
Comment (ajanthan) [reply apm-collector/apm-collector-storage/collector-storage-es-provider/src/main/java/org/apache/skywalking/apm/collector/storage/es/dao/ui/SegmentDurationEsUIDAO.java]: It is a mistake. Fixed it in the latest pull request.

File: apm-collector/apm-collector-storage/collector-storage-es-provider/src/main/java/org/apache/skywalking/apm/collector/storage/es/dao/ui/SegmentDurationEsUIDAO.java [hunk 2]
```diff
@@ -73,16 +75,30 @@ public TraceBrief loadTop(long startSecondTimeBucket, long endSecondTimeBucket,
...
             boolQueryBuilder.must().add(QueryBuilders.termsQuery(SegmentDurationTable.SEGMENT_ID.getName(), segmentIds));
         }
         if (applicationId != 0) {
             boolQueryBuilder.must().add(QueryBuilders.termQuery(SegmentDurationTable.APPLICATION_ID.getName(), applicationId));
         }
-
-        searchRequestBuilder.addSort(SegmentDurationTable.START_TIME.getName(), SortOrder.DESC);
+        switch (traceState) {
+            case ERROR:
+                mustQueryList.add(QueryBuilders.matchQuery(SegmentDurationTable.IS_ERROR.getName(), 1));
```
[the remark this answers is not in the document]
Comment (ajanthan) [reply apm-collector/apm-collector-storage/collector-storage-es-provider/src/main/java/org/apache/skywalking/apm/collector/storage/es/dao/ui/SegmentDurationEsUIDAO.java]: Changed it.

File: apm-collector/apm-collector-storage/collector-storage-es-provider/src/main/java/org/apache/skywalking/apm/collector/storage/es/dao/ui/SegmentDurationEsUIDAO.java [hunk 3]
```diff
@@ -73,16 +75,30 @@ public TraceBrief loadTop(long startSecondTimeBucket, long endSecondTimeBucket,
...
+                break;
+            case SUCCESS:
+                mustQueryList.add(QueryBuilders.matchQuery(SegmentDurationTable.IS_ERROR.getName(), 0));
```
[the remark this answers is not in the document]
Comment (ajanthan) [reply apm-collector/apm-collector-storage/collector-storage-es-provider/src/main/java/org/apache/skywalking/apm/collector/storage/es/dao/ui/SegmentDurationEsUIDAO.java]: Changed it.

**Pull request merged**